api/pwdb.proto

```diff
@@ -180,23 +179,27 @@ message OrganizationMember {
 }
 
 message TeamMember {
-  int64 id = 1 [(gogoproto.moretags) = "gorm:\"primary_key\"", (gogoproto.customname) = "ID"];
+  int64 id = 1 [(gogoproto.moretags) = "gorm:\"primaryKey\"", (gogoproto.customname) = "ID"];
   google.protobuf.Timestamp created_at = 2 [(gogoproto.stdtime) = true, (gogoproto.nullable) = true];
   google.protobuf.Timestamp updated_at = 3 [(gogoproto.stdtime) = true, (gogoproto.nullable) = true];
 
   Role role = 100;
   string slug = 102;
 
-  User user = 200 [(gogoproto.moretags) = "gorm:\"foreignkey:UserID\""];
-  int64 user_id = 201 [(gogoproto.customname) = "UserID", (gogoproto.moretags) = "sql:\"not null\" gorm:\"index\""];
-  Team team = 202 [(gogoproto.moretags) = "gorm:\"foreignkey:TeamID\""];
-  int64 team_id = 203 [(gogoproto.customname) = "TeamID", (gogoproto.moretags) = "sql:\"not null\" gorm:\"index\""];
-  repeated Coupon coupons = 204 [(gogoproto.moretags) = "gorm:\"PRELOAD:false\""];
-  repeated Achievement achievements = 205 [(gogoproto.moretags) = "gorm:\"PRELOAD:false\""];
-  repeated WhoswhoAttempt sent_whoswho_attempts = 206 [(gogoproto.moretags) = "gorm:\"PRELOAD:false\""];
-  repeated WhoswhoAttempt received_whoswho_attempts = 207 [(gogoproto.moretags) = "gorm:\"PRELOAD:false\""];
-  repeated InventoryItem items = 208 [(gogoproto.moretags) = "gorm:\"PRELOAD:false\""];
-  repeated ChallengeValidation challenge_validations = 209 [(gogoproto.moretags) = "gorm:\"PRELOAD:false\""];
+  // belongs to
+  int64 user_id = 200 [(gogoproto.customname) = "UserID", (gogoproto.moretags) = "gorm:\"index\""];
+  User user = 201 [(gogoproto.moretags) = "gorm:\"foreignKey:UserID;references:ID\""];
+  // has one
+  User active_user = 202 [(gogoproto.moretags) = "gorm:\"foreignKey:ActiveTeamMemberID;references:ID\""];
```

Comment:
ForeignKey constraint fails here -> need to investigate or 
maybe need to rethink the model association and use a enum to mark a team_member as the active one ?